### PR TITLE
feat: add resources to open-api sidebar

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds an optional `Resources` section to the OpenApiDocsView sidebar.

## 📸 Design Screenshots

| Before | After |
| - | - |
| before | after |

## 🧪 Testing

- [ ] Visit [/hcp/api-docs/vault-secrets]
   - In the sidebar, a `Resources` section should appear below the items parsed from the OpenAPI spec
   - This `Resources` section should also appear in the mobile menu

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1204678746647847/1205295914128349/f
[preview]: https://dev-portal-git-zsopenapi-sidebar-resources-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zsopenapi-sidebar-resources-hashicorp.vercel.app/hcp/api-docs/vault-secrets

### CODE REVIEW

**Comment by zchsh** [comment]
@braydenlove This PR adds a `Resources` section to the bottom of the OpenAPIView sidebar. It's optional, so we can choose to include this or not on a case-by-case basis 👍

File: src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx [hunk 1]
```diff
@@ -51,6 +54,28 @@ const PRODUCT_SLUG = 'hcp'
  */
 const BASE_URL = '/hcp/api-docs/vault-secrets'
 
+/**
+ * Resource items are shown in the sidebar
+ */
+const NAV_RESOURCE_ITEMS: OpenApiNavItem[] = [
```
Comment (kendallstrautman) [inline src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx:60]: 💡 I wonder if a `constants` file might be helpful for all these config values.

File: src/views/open-api-docs-view/components/open-api-sidebar-contents/index.tsx [hunk 2]
```diff
@@ -75,9 +77,30 @@ export function OpenApiSidebarContents({
 	}, [navItems, activeSection, urlPathname])
 
 	// Render a generic list of `SideBarNavMenuItem`
+	return (
+		<>
+			<SidebarNavMenuItemsList items={navItemsWithActive} />
+			{/* Render resources, if present */}
+			{navResourceItems?.length > 0 ? (
```
Comment (kendallstrautman) [inline src/views/open-api-docs-view/components/open-api-sidebar-contents/index.tsx:84]: Much more flexible 💯

**Review by kendallstrautman — APPROVED** [review]

File: src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx (hunk 1 above)
Comment (zchsh) [reply src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx:60]: ooh i like that a lot! will add in a follow-up PR 👍

**Pull request merged**